xformers/components/attention/scaled_dot_product.py

```diff
@@ -97,19 +97,20 @@ def forward(
             )
 
         # Handle a possibly deferred causal mask handling
+        mask = self.mask
```

Comment:
@erip there's something wrong in the logic here, looks like self.mask is never set, is it ? I'm checking that it's not set elsewhere, but it feels like the caching is gone, which is a shame if the same seq length repeats.

Reply:
ok, I just got a little more context with the linked issue, not super clear that caching the mask is always a win indeed. @fmassa wanted to rewrite all the mask handling at some point, it would make sense I think. 

A 12 layers 4k context model would carry over 400MB worth of masks (fp16) if using the self.mask member here, maybe that caching is not such a great idea vs. passing a matching att mask all the time for instance